articles/ros2_threat_model.md

```diff
@@ -2492,6 +2495,7 @@ enable_discovery_protection to True to prevent malicious actors from
 enumerating and fingerprinting DDS participants.</li>
         <li>DDS Security Governance document must enable_liveliness_protection
 to True</li>
+        <li>XRCE agent to LIDAR should use secure UDP protocol like DTLS</li>
```

Comment:
Is this actually do-able as of today? Or is this a suggestion for the spec?

Reply:
This is a suggestion for the spec, as far as I can tell only there are only implementations for UDP, TCP, and serial transport layers under DDS-XRCE. I've tried to clarify that in the doc and given another suggestion for securing the UDP communication. Looking at the LIDAR source code for the turtlebot (https://github.com/ROBOTIS-GIT/turtlebot3/blob/ros2/turtlebot3_lidar/turtlebot3_lidar/main.cpp) it's just communicating with localhost so I'm not sure what general network security recommendations are applicable here.

Reply:
The main problem I see with that is that this doc needs to highlight actionable steps and we cannot modify the spec. What about recommending instead:
1/ never use XRCE for communicating over an insecure transport layer (ethernet unless one is using e.g. a VPN tunnel).
2/ If communication is only done locally, it should be done over the loopback interface.

Reply:
For the Turtlebot specific threats I've removed mitigations that aren't immediately actionable. I think these recommendations are already covered in the more general ROS2 model with mitigations like "communicating over an encrypted channel"